### PR TITLE
fix readme regarding http option, make getHttpOpts resilient against null parameter

### PR DESCRIPTION
**Pull request by Uzlopak**
follow up to #267  and #266 

#### Checklist

- [ ] run `npm run test` and `npm run benchmark`
- [ ] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [ ] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: lib/request.js [hunk 1]
```diff
@@ -253,7 +253,7 @@ function getHttp2Opts (opts) {
 }
 
 function getHttpOpts (opts) {
-  const httpOpts = typeof opts.http === 'object' ? opts.http : {}
+  const httpOpts = typeof opts.http === 'object' && opts.http !== null ? opts.http : {}
```
Comment (mcollina) [inline lib/request.js:256]: can you add a test for this?

Comment (Uzlopak) [reply lib/request.js:256]: yeah definetly. seems like there are some more issues.

**Comment by RafaelGSS** [comment]
Honestly, I'd follow the `http2` use case https://github.com/fastify/fastify-reply-from/pull/267#issuecomment-1212102941.

IMO It's way clear

**Review by mcollina — APPROVED** [review]
lgtm

File: lib/request.js (hunk 1 above)
Comment (Uzlopak) [reply lib/request.js]: I removed that check, as it seems that null is not the same result as when having an empty object. I assume that it has something to do with how agentOptions are affected.

**Comment by Uzlopak** [comment]
@RafaelGSS 

I think that would result in a breaking change.

**Pull request merged**